app/test/features/chat_ng/chat_editor_test.dart

```diff
@@ -0,0 +1,313 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/sdk_provider.dart';
+import 'package:acter/features/chat_ng/models/chat_editor_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:acter/features/chat_ng/providers/notifiers/chat_editor_notifier.dart';
+import 'package:acter/features/chat_ng/widgets/chat_editor/chat_editor.dart';
+import 'package:acter/features/chat_ng/widgets/chat_editor/chat_editor_actions_preview.dart';
+import 'package:acter/features/chat_ng/widgets/message_actions_widget.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../helpers/mock_a3sdk.dart';
+import '../../helpers/test_wrapper_widget.dart';
+import '../comments/mock_data/mock_message_content.dart';
+import 'diff_applier_test.dart';
+import 'messages/chat_message_test.dart';
+
+class MockChatEditorNotifier extends AutoDisposeNotifier<ChatEditorState>
+    with Mock
+    implements ChatEditorNotifier {
+  @override
+  ChatEditorState build() => ChatEditorState(
+        selectedMsgItem: null,
+        actionType: MessageAction.none,
+      );
+}
+
+void main() {
+  group('Chat editor reply/edit preview tests', () {
```

Comment:
And finally some widgets tests to ensure that actions state logics are working as intended!